sandbox/components/PageHeader.vue

```diff
@@ -36,7 +40,19 @@ const query = ref('')
 
 watch(query, (searchQuery: string) => {
   emit('search', searchQuery)
+
+  if (searchQuery) {
```

Comment:
You need to add a check here to make sure the query doesn't already exist so that it doesn't continuously route

Reply:
Ah fair enough

Reply:
Warning: making mistakes in code can summon Adam any time of day

Reply:
Just like to help 😅 (in an Uber trying to ignore the music lol)

Reply:
Fixed

Reply:
This is all it requires:

```
watch(query, (searchQuery: string) => {
  emit('search', searchQuery)

  if (searchQuery) {
    router.push({ name: 'home', query: { search: searchQuery } })
  }
}, { immediate: true })
```
It's already checking if the query exists and then redirects them to the correct query param. I don't see a potential infinite loop 🤔

Reply:
It's going to push every time the query string changes, which will also make the component remount every time. At a minimum it needs a debounce on the search string

Reply:
yes, debounce is a valid point. 

If we support search through query param, it needs to be reflected on the url too

Reply:
If I didn't have `else` block it doesn't remove the query when search is empty (recording)

https://github.com/Kong/icons/assets/36751160/5ae7617d-0d57-486d-bbf1-5a321991eb87

Reply:
<img width="1010" alt="Screenshot 2023-09-20 at 4 05 27 PM" src="https://github.com/Kong/icons/assets/6960750/95b2ce0a-1f16-45f1-acf3-f640af40fda4">

Understood, what I was saying is the query param is lost when we share the link and opens in a window. 

Anyways, this is not a deal breaker in any ways. We can fix it later if it's an issue.

Reply:
Yes, but I would say don't merge this sandbox change if it's not completely functional 🙏